library/std/src/net/addr.rs

```diff
@@ -364,7 +364,6 @@ impl SocketAddrV6 {
                 sin6_addr: *ip.as_inner(),
                 sin6_flowinfo: flowinfo,
                 sin6_scope_id: scope_id,
-                ..unsafe { mem::zeroed() }
```

Comment:
This is surprising. Which Clippy lint was this? Could there be a platform where there are other fields?

Reply:
Ah, `needless_update`. The other question remains.

Reply:
Took the liberty to do some digging. `sockaddr_in6` is defined in-tree for some platforms and in the libc crate for others. The in-tree ones all have the fields listed here but libc has more variety. BSD, Haiku, and a few others have an additional `sin6_len: u8` field. Solaris and Illumos have `__sin6_src_id: u32` instead.

Reply:
I'm assuming feature gating this would be messy, and I should just revert that change?

Reply:
Reverting it seems reasonable.  Then the other changes can go in easily.

Reply:
done